akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorSystemAdapter.scala

```diff
@@ -129,15 +128,32 @@ import akka.annotation.InternalApi
 }
 
 private[akka] object ActorSystemAdapter {
-  def apply(system: classic.ActorSystem): ActorSystem[Nothing] = AdapterExtension(system).adapter
+  def apply(system: classic.ActorSystem): ActorSystem[Nothing] = {
+    system match {
+      case system: ActorSystemImpl =>
+        // Optimization to cut out going through adapter lookup if possible
+        system.typedSystem match {
+          case OptionVal.Some(typedSystem: ActorSystem[_]) => typedSystem
+          case _ =>
+            val typedSystem = AdapterExtension(system).adapter
+            system.typedSystem = OptionVal.Some(typedSystem)
```

Comment:
Wonder if something is wrong here? Here it sets it to `ActorSystemAdapter`, but above it looks for `: ActorSystem[_]`?

Reply:
`ActorSystemAdapter` is the (adapted) typed system. Would be fine to be explicit about it and match on the adapter type though.

Reply:
ah, ofc